### ISSUE TITLE
CrashLoopBackOff in OpenShift environment

### ISSUE THREAD

**Issue by kevinleturc** [DataDog/datadog-agent#11131]
**Describe what happened:**

Since `7.34.0`, the docker image of the agent is crashing at each startup.
See below the full logs:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
--
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/50-ci.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/01-check-apikey.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/50-ecs.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/50-eks.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/50-kubernetes.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/50-mesos.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/51-docker.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/60-network-check.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/59-defaults.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/60-sysprobe-check.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/89-copy-customfiles.sh: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/agent/run: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/process/run: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/sysprobe/run: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/trace/run: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/security/run: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/agent/finish: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/process/finish: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/security/finish: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/sysprobe/finish: Operation not permitted
  | s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/trace/finish: Operation not permitted
  | [s6-init] ensuring user provided files have correct perms...exited 0.
  | [fix-attrs.d] applying ownership & permissions fixes...
  | [fix-attrs.d] done.
  | [cont-init.d] executing container initialization scripts...
  | [cont-init.d] 01-check-apikey.sh: executing...
  | [cont-init.d] 01-check-apikey.sh: exited 0.
  | [cont-init.d] 50-ci.sh: executing...
  | [cont-init.d] 50-ci.sh: exited 0.
  | [cont-init.d] 50-ecs.sh: executing...
  | [cont-init.d] 50-ecs.sh: exited 0.
  | [cont-init.d] 50-eks.sh: executing...
  | [cont-init.d] 50-eks.sh: exited 0.
  | [cont-init.d] 50-kubernetes.sh: executing...
  | [cont-init.d] 50-kubernetes.sh: exited 0.
  | [cont-init.d] 50-mesos.sh: executing...
  | [cont-init.d] 50-mesos.sh: exited 0.
  | [cont-init.d] 51-docker.sh: executing...
  | [cont-init.d] 51-docker.sh: exited 0.
  | [cont-init.d] 59-defaults.sh: executing...
  | [cont-init.d] 59-defaults.sh: exited 0.
  | [cont-init.d] 60-network-check.sh: executing...
  | [cont-init.d] 60-network-check.sh: exited 0.
  | [cont-init.d] 60-sysprobe-check.sh: executing...
  | grep: /etc/datadog-agent/system-probe.yaml: No such file or directory
  | grep: /etc/datadog-agent/system-probe.yaml: No such file or directory
  | [cont-init.d] 60-sysprobe-check.sh: exited 0.
  | [cont-init.d] 89-copy-customfiles.sh: executing...
  | /conf.d/..2022_03_02_15_09_45.056178133 -> /etc/datadog-agent/conf.d/..2022_03_02_15_09_45.056178133
  | '/conf.d/..2022_03_02_15_09_45.056178133/conf.yaml' -> '/etc/datadog-agent/conf.d/..2022_03_02_15_09_45.056178133/conf.yaml'
  | '/conf.d/conf.yaml' -> '/etc/datadog-agent/conf.d/conf.yaml'
  | [cont-init.d] 89-copy-customfiles.sh: exited 0.
  | [cont-init.d] done.
  | [services.d] starting services
  | starting security-agent
  | starting process-agent
  | starting agent
  | starting trace-agent
  | starting system-probe
  | [services.d] done.
  | 2022-03-02 15:18:13 UTC \| PROCESS \| INFO \| (pkg/util/log/log.go:572 in func1) \| Features detected from environment: kubernetes
  | 2022-03-02 15:18:13 UTC \| PROCESS \| INFO \| (pkg/util/log/log.go:572 in func1) \| no config exists at /etc/datadog-agent/system-probe.yaml, ignoring...
  | 2022-03-02 15:18:13 UTC \| PROCESS \| INFO \| (pkg/util/log/log.go:572 in func1) \| overriding API key from env DD_API_KEY value
  | 2022-03-02 15:18:15 UTC \| CORE \| INFO \| (pkg/util/log/log.go:572 in func1) \| runtime: final GOMAXPROCS value is: 1
  | 2022-03-02 15:18:15 UTC \| CORE \| INFO \| (pkg/util/log/log.go:572 in func1) \| Features detected from environment: kubernetes
  | 2022-03-02 15:18:15 UTC \| CORE \| INFO \| (cmd/agent/app/run.go:249 in StartAgent) \| Starting Datadog Agent v7.34.0
  | 2022-03-02 15:18:16 UTC \| CORE \| INFO \| (cmd/agent/app/run.go:318 in StartAgent) \| Hostname is: arender-pool-1-document-text-handler-3-r6gc8
  | 2022-03-02 15:18:20 UTC \| CORE \| ERROR \| (pkg/api/security/security.go:248 in saveAuthToken) \| Failed to write auth token acl couldn't set user and group owner for /etc/datadog-agent/auth_token: chown /etc/datadog-agent/auth_token: operation not permitted
  | 2022-03-02 15:18:20 UTC \| CORE \| ERROR \| (cmd/agent/app/run.go:340 in StartAgent) \| Error while starting api server, exiting: error writing authentication token file on fs: couldn't set user and group owner for /etc/datadog-agent/auth_token: chown /etc/datadog-agent/auth_token: operation not permitted
  | 2022-03-02 15:18:20 UTC \| CORE \| INFO \| (pkg/logs/logs.go:176 in Stop) \| Stopping logs-agent
  | 2022-03-02 15:18:20 UTC \| CORE \| INFO \| (pkg/logs/logs.go:188 in Stop) \| logs-agent stopped
  | 2022-03-02 15:18:20 UTC \| CORE \| INFO \| (cmd/agent/app/run.go:501 in StopAgent) \| See ya!
  | Error: Error while starting api server, exiting: error writing authentication token file on fs: couldn't set user and group owner for /etc/datadog-agent/auth_token: chown /etc/datadog-agent/auth_token: operation not permitted
  | AGENT EXITED WITH CODE 255, SIGNAL 0, KILLING CONTAINER
  | process-agent exited with code 256, signal 15, restarting in 2 seconds
  | security-agent exited with code 256, signal 15, restarting in 2 seconds
  | system-probe exited with code 256, signal 15, restarting in 2 seconds
  | [cont-finish.d] executing container finish scripts...
  | [cont-finish.d] done.
  | [s6-finish] waiting for services.
  | 2022-03-02 15:18:22 UTC \| TRACE \| INFO \| (pkg/util/log/log.go:572 in func1) \| runtime: final GOMAXPROCS value is: 1
  | 2022-03-02 15:18:22 UTC \| TRACE \| INFO \| (pkg/util/log/log.go:572 in func1) \| Features detected from environment: kubernetes
  | 2022-03-02 15:18:22 UTC \| TRACE \| INFO \| (pkg/util/log/log.go:572 in func1) \| Loaded configuration: /etc/datadog-agent/datadog.yaml
  | 2022-03-02 15:18:22 UTC \| TRACE \| INFO \| (pkg/util/log/log.go:567 in func1) \| Activating non-local traffic automatically in containerized environment, trace-agent will listen on 0.0.0.0
  | 2022-03-02 15:18:22 UTC \| TRACE \| INFO \| (pkg/util/log/log.go:572 in func1) \| received signal 15 (terminated)
  | 2022-03-02 15:18:22 UTC \| TRACE \| INFO \| (pkg/tagger/remote/tagger.go:333 in func1) \| unable to establish stream, will possibly retry: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp :5001: connect: connection refused"
  | s6-svwait: fatal: supervisor died
  | 2022-03-02 15:18:22 UTC \| TRACE \| INFO \| (pkg/tagger/remote/tagger.go:333 in func1) \| unable to establish stream, will possibly retry: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp :5001: connect: connection refused"
  | [s6-finish] sending all processes the TERM signal.
  | trace-agent exited with code 256, signal 1, restarting in 2 seconds
  | [s6-finish] sending all processes the KILL signal and exiting.
```

**Describe what you expected:**
The datadog agent is starting correctly.

**Steps to reproduce the issue:**
Just start the container in an OpenShift environment.

**Additional environment details (Operating System, Cloud provider, etc):**
OpenShift cluster v3.9.0+2e78773-56 installed on AWS EC2.

**Comment by vboulineau** [comment]
Hello @kevinleturc,

You mention that this started as `7.34`, does it mean it worked with `7.33` or were you using an older version previously?
Also you could share the SCC used for this POD?

**Comment by chenyang-nirovision** [comment]
We encountered a similar issue on `7.34.0` after noticing Datadog agent didn't publish `eks.fargate.pods.running` metrics (We run datadog agent as a side car container on AWS EKS Fargate environment)
```
2022-03-04 00:20:30 UTC | TRACE | WARN | (pkg/tagger/remote/tagger.go:239 in run) | error received from remote tagger: rpc error: code = DeadlineExceeded desc = timed out waiting for operation
```
We rolled back to version `7.33.1` it works.

**Comment by kevinleturc** [comment]
Hello @vboulineau,
Indeed the `7.33.1` version is working correctly, we fixed this version to have a successful deployment.

The SCC of the pod is _restricted_, which I believe is the default SCC of OpenShift.

**Comment by nicolamarangoni** [comment]
Hello, I have the same issue but I install the agent using the script:
https://s3.amazonaws.com/dd-agent/scripts/install_script.sh
However, this script allow to define just the major version.
Is there a script to quickly install a specific version, like` 7.33.1`

**Comment by freshuk** [comment]
same here..how do we downgrade to 7.33.1 as the new version gives us an error:
error received from remote tagger: rpc error: code = Unavailable desc = transport is closing

**Comment by bheemreddy181** [comment]
same issue here

**Comment by chiragkanhasoft** [comment]
I have used `datadog/agent:latest` and it was generating an error. So now i change it to the `datadog/agent:7.33.1` and it starts working again. But now there is no data in APM.

**Comment by mattbookpr0** [comment]
For others that come across this thread, I had very similar looking logs with auth errors and crashloops running version `7.40.1` but I was able to observe the pod CrashLoopBackOff in GKE briefly mention OOMKill during the restarts. I increased memory allocation to the DD agent pod and now it is running smoothly.

**Comment by lpodolak** [comment]
> For others that come across this thread, I had very similar looking logs with auth errors and crashloops running version `7.40.1` but I was able to observe the pod CrashLoopBackOff in GKE briefly mention OOMKill during the restarts. I increased memory allocation to the DD agent pod and now it is running smoothly.

Yes, increasing resource limits for my datadog agent solved the problem as well in version 7.70.2
